modules/imgcodecs/src/loadsave.cpp

```diff
@@ -1198,6 +1198,8 @@ bool imencode( const String& ext, InputArray _image,
 
 bool haveImageReader( const String& filename )
 {
+    if (haveImageWriter(filename))
+        return true;
```

Comment:
Why writer?

Reply:
`cv::haveImageReader(".jpg")` - need to be fixed.
`haveImageWriter(filename)` some formats may have reader only.